pinot-core/src/main/java/com/linkedin/pinot/core/operator/docvaliterators/RealtimeSingleValueIterator.java

```diff
@@ -71,6 +71,36 @@ public int nextIntVal() {
   }
 
   @Override
+  public long nextLongVal() {
+    if (!hasNext()) {
+      return Constants.EOF;
+    }
+
+    long ret = reader.getLong(counter);
+    return ret;
```

Comment:
missing the counter increment in nextLong(), nextFloat(), nextDouble(). Will fix, and try to add a unit test as well